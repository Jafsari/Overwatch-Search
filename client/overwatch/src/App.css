App {
  text-align: center;
  background-image: linear-gradient(rgba(0,0,0,0.7), rgba(0,0,0,0.7)),url(pics/bridgette.jpeg);
  height:100vh;
  background-size:cover;
  background-position:center;
  background-attachment: fixed;
}

.SearchBackground {
  text-align: center;
  background-image: url(pics/reinhardt-shield.jpg);
  height:100vh;
  background-size:cover;
  background-position:center;
  background-attachment: fixed;
}

.GChat {
  text-align: center;
  background-image: url(pics/roadhog-junk.jpg);
  height:100vh;
  background-size:cover;
  background-position:center;
  background-attachment: fixed;
  display:flex;
  flex-direction:column;
  justify-content:center;
}

.inviteBackground{
  text-align: center;
  background-image: linear-gradient(rgba(0,0,0,0.4), rgba(0,0,0,0.4)),url(pics/bastion.jpg);
  height:100vh;
  background-size:cover;
  background-position:center;
  background-attachment: fixed;
}
.col-8{
  position:absolute;
  left:175px;
}

.CChat {
  text-align: center;
  background-image: url(pics/bridgette.jpeg);
  height:100vh;
  background-size:cover;
  background-position:center;
  background-attachment: fixed;
  display:flex;
  flex-direction:column;

}
.DiaChat {
  text-align: center;
  background-image:url(pics/anime-tracer.jpg);
  height:100vh;
  background-size:cover;
  background-position:center;
  background-attachment: fixed;
  display:flex;
  flex-direction:column;
}
.MasChat {
  text-align: center;
  background-image: url(pics/junkrat.jpg);
  height:100vh;
  background-size:cover;
  background-position:center;
  background-attachment: fixed;
  display:flex;
  flex-direction:column;

}
.GrandMasChat {
  text-align: center;
  background-image: url(pics/Dva.jpg);
  height:100vh;
  background-size:cover;
  background-position:center;
  background-attachment: fixed;
    display:flex;
  flex-direction:column;

}


.Dashboard2{
  text-align: center;
  background-image:linear-gradient(rgba(0,0,0,0.3), rgba(0,0,0,0.3)),url(pics/overwatch-league.jpg);
  height:100vh;
  background-size:cover;
  background-position:center;
  background-attachment: fixed;
}

.Dash{
  text-align: center;
  background-image: linear-gradient(rgba(0,0,0,0.4), rgba(0,0,0,0.4)),url(pics/winstonEarth.jpg);
  height:100vh;
  background-size:cover;
  background-position:center;
  background-attachment: fixed;
}

.signup{
  text-align: center;
  background-image: linear-gradient(rgba(0,0,0,0.7), rgba(0,0,0,0.7)),url(pics/overwatchTeam.jpg);
  height:100vh;
  background-size:cover;
  background-position:center;
  background-attachment: fixed;
}

.login{
    text-align: center;
    background-image: url(pics/overwatch-serious.jpg);
    height:100vh;
    background-size:cover;
    background-position:center;
    background-attachment: fixed;
}

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 80px;
}

.App-header {
  background-color: #222;
  height: 150px;
  padding: 20px;
  color: white;
}

.App-title {
  font-size: 1.5em;
}

.App-intro {
  font-size: large;
}

.format {
  display:flex;
  flex-direction:column;
  position: fixed;
  top: 31%;
  left: 39%;
  margin-top: -50px;
  margin-left: -100px;
  color:white;
  
}

.card-body{
  overflow:scroll;
}
.dashboard{
  display:flex;
  flex-direction:column;
  position: fixed;
  top: 15%;
  left: 40%;
  margin-top: -50px;
  margin-left: -100px;
  color:white;
  /* position: absolute;
  left:500px;
  top:50px;
  color: white; */
}
.login2{
  display:flex;
  flex-direction:column;
  position: fixed;
  top: 31%;
  left: 49%;
  margin-top: -50px;
  margin-left: -100px;
  color:white;
  /* position: absolute;
  left:610px;
  top:200px;
  color: white; */

}

.Searchlayout{
  display:flex;
  flex-direction:column;
  position:fixed;
  top: 25%;
  left: 36%;
  color:white;
  /* position: absolute;
  left:475px;
  top:200px;
  color: white; */

}

.inviteLoading{
  display:flex;
  flex-direction:column;
  position:fixed;
  top: 38%;
  left: 47%;
  color:white;
}
.chat{
  /* display:flex;
  flex-direction:column;
  justify-content:center; */
  /* display:flex;
  flex-direction:column; */

}

.inline{
  display:flex;
  justify-content:center;

}

@keyframes App-logo-spin {
  from { transform: rotate(0deg); }
  to { transform: rotate(360deg); }
}

.card-body{
  height:300px;
}

.col-8{
  position:relative;
  top:-55px;
}

.message{
  display:flex;
}

.grandmaster{
/* position:absolute;
top:-340px;
right:700px; */
height:225px;
width:225px;
position:relative;
left:-15px;
-moz-transform:scale(0.5);
-webkit-transform:scale(0.5);
transform:scale(0.5);
}

.SearchSpace{
display:block;
margin-bottom:20px;
}

.EmailSpace{
  display:block;
  height:100px;
  margin-bottom:20px;
}

.SearchMutton{
/* position:relative;
right:10px;
bottom:10px; */
}

.charmander{
  color:red;
/* position:relative;
right:10px;
bottom:10px; */
}



.master{
/* position:absolute;
top:-275px;
right:780px; */
position:relative;
height:225px;
width:225px;
left:-15px;
-moz-transform:scale(0.7);
-webkit-transform:scale(0.7);
transform:scale(0.7);
}

.diamond{
  /* position:absolute;
  top:-275px;
  right:780px; */
height:225px;
width:225px;
position:relative;
left:-15px;
  -moz-transform:scale(0.7);
  -webkit-transform:scale(0.7);
  transform:scale(0.7);
  }

  .diamondlogo{
    padding:-105px;
    margin:-118px;
    -moz-transform:scale(0.1);
    -webkit-transform:scale(0.1);
    transform:scale(0.1);
  }

  .complogo{
    background:black;
    padding:-105px;
    margin:-90px;
    -moz-transform:scale(0.1);
    -webkit-transform:scale(0.1);
    transform:scale(0.1);
  }

  .analogo{
    background:black;
    padding:-105px;
    margin:-5px;
    -moz-transform:scale(0.1);
    -webkit-transform:scale(0.1);
    transform:scale(0.1);
  }

  .grandlogo{
    position:relative;
    bottom:4px;
    padding:-105px;
    margin:-185px;
    -moz-transform:scale(0.08);
    -webkit-transform:scale(0.08);
    transform:scale(0.08);
  }

  .masterlogo{
    position:relative;
    bottom:2px;
    padding:-105px;
    margin:-115px;
    -moz-transform:scale(0.1);
    -webkit-transform:scale(0.1);
    transform:scale(0.1);
  }


  .competitiveLogo{
    /* position:absolute;
    top:50px;
    right:800px; */
    position:relative;
    bottom:30px;
    height:225px;
    left:-15px;
    -moz-transform:scale(0.5);
    -webkit-transform:scale(0.5);
    transform:scale(0.5);
    }

    .Reinhardt{
      position:fixed;
      bottom:265px;
      right:200px;
      -moz-transform:scale(0.2);
      -webkit-transform:scale(0.2);
      transform:scale(0.2);
      }

      .chatNavigation{
        position: absolute;
        left:320px;
        bottom:187px;
        color: black;
        width:50.7%
      }
      .compNavigation{
        /* display:flex;
        position:relative;
        bottom:22.5%;
        right:39%; */
        
        position: relative;
        left:165px;
        bottom:60px;
        color: black;
        width:50.7%
      }
      .masterNavigation{
        position: absolute;
        left:165px;
        bottom:1px;
        color: black;
        width:50.7%
      }
      .GMNavigation{
        position: absolute;
        left:165px;
        bottom:1px;
        color: black;
        width:50.7%
      }
      .diamondNavigation{
        position: absolute;
        left:165px;
        bottom:1px;
        color: black;
        width:50.7%
      }

      .player{
        position:relative;
        left:30px;
        float:left;
      }
      .player-card{
        position:relative;
        right:50px;
      }
      .error {
        color: red;
        font-size: 12px;
        margin-top: 5px;
      }
      
