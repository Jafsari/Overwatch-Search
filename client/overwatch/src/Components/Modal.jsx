import React from 'react';
import { Button, Modal, ModalHeader, ModalBody, ModalFooter } from 'reactstrap';

class ModalExample extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      modal: false
    };

    this.toggle = this.toggle.bind(this);
  }

  toggle() {
    this.setState({
      modal: !this.state.modal
    });
  }

  render() {
    return (
      <div>
        <Button color="danger" onClick={this.toggle}>{this.props.buttonLabel} More Info</Button>
        <Modal isOpen={this.state.modal} toggle={this.toggle} className={this.props.className}>
          <ModalHeader toggle={this.toggle}>BojiWatch</ModalHeader>
          <ModalBody>
            BojiWatch is an web application that brings competitive overwatch players together. Find other serious players to play competitive ranked matches with! Best of luck!
          </ModalBody>
          <ModalFooter>
            <Button color="primary" onClick={this.toggle}>Thanks!</Button>{' '}
            <Button color="secondary" onClick={this.toggle}>Cancel</Button>
          </ModalFooter>
        </Modal>
      </div>
    );
  }
}

export default ModalExample;