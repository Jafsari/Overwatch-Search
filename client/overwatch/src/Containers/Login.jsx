import { connect } from 'react-redux';
import * as actions from '../actions'
import React, { Component } from 'react';
import { Button, FormGroup, Label, Input } from 'reactstrap';
import '../App.css';

class Signup extends Component{
    constructor(props){
        super(props)
        this.state = {
            username:"",
            password:""
        }
    }

    componentWillMount(){
        if (localStorage.jwtToken !== undefined) {
           this.props.history.push('dashboard')
        }
    }
    clear = () => {
        this.setState({
            username:"",
            password:""
        });
    }
    handleChange = (e) => {
        this.setState({
            [e.target.name]: e.target.value
          });
        
    }
    handleSubmit = (e) => {
        e.preventDefault()
        console.log(this.state)
        this.props.login(this.state).then(
            () => {
                this.props.history.push('dashboard')
            },
            err => {
                console.log(err.message)
            }
        );
        this.props.setCurrentUser(this.state.username)
            this.clear();
            console.log(`You're in!`)
    }

    render(){
        return (
            <div className="login">
            <form className="login2">
              <div>
        <FormGroup onSubmit={this.handleSubmit}>
          <Label for="exampleEmail">Username</Label>
          <Input     
          onChange={this.handleChange}
          placeholder="Enter username"
          name="username"
          type="text"
          id="username"
          value={this.state.username} />
        </FormGroup>
        <FormGroup>
          <Label for="examplePassword">Password</Label>
          <Input 
                    onChange={this.handleChange}
                    placeholder="Enter password"
                    name="password"
                    id="password"
                    type="password"
                    value={this.state.password}/>
        </FormGroup>
      </div>
            <div>
            <Button onClick ={this.handleSubmit} color="danger">Login</Button>
            </div>
            </form>
            </div>
        
        )
    }
}

export default connect(null,actions)(Signup);