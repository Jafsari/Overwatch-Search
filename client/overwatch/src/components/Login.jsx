import { connect } from 'react-redux';
import React, { Component } from 'react';
import '../App.css';
import BasicForm from '../Containers/Form.jsx';


class Login extends Component{
    constructor(props){
        super(props)
    }
    componentDidMount(){
    if (this.props.isAuthenticated){
        this.props.history.push('dashboard')
    }
}

render(){
        return (
            <div className="login">
            <form className="login2">
              <div>
         <BasicForm current='Login'/>
      </div>
            <div>
            </div>
            </form>
            </div>
        
        )
    }
}
const mapStateToProps = (state) => { 
    return { 
      user: state.auth.user ,
      isAuthenticated: state.auth.isAuthenticated
      };
  };
  
  export default connect(mapStateToProps,null)(Login);
