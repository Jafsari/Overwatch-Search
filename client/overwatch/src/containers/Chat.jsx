import React from "react";
import io from "socket.io-client";
import '../App.css'
import { connect } from 'react-redux';
import Navigation from '../components/Navigation.jsx';
import Infinite from 'react-infinite';
import { Alert } from 'reactstrap';


class Chat extends React.Component{
    constructor(props){
        super(props);

        this.state = {
            username: this.props.user.toString(),
            message: '',
            messages: []
        };

          this.socket = io(this.props.server);
          console.log(this.state.messages)
        this.socket.on('RECEIVE_MESSAGE', function(data){
            addMessage(data);
        });

        const addMessage = data => {
            console.log(data);
            this.setState({messages: [...this.state.messages, data]});
            console.log(this.state.messages);
        };

        this.sendMessage = ev => {
            ev.preventDefault();
            console.log(this.state.message.length)
            if (this.state.message.length === 0){
                return;
            } else {
            this.socket.emit('SEND_MESSAGE', {
                author: this.state.username,
                message: this.state.message
            })
            this.setState({message: ''});

        }
    }

    }

    render(){
        
        return (
          <div className="chat">
          <form onSubmit={this.sendMessage} >
             <img src={this.props.image} className={this.props.imageStyle} alt="logo" />
            <div className="container">
                <div className="row">
                    <div className="col-8">
                        <div className="card">
                            <div className="card-body">
                                <div className="card-title"><strong>{this.props.name}</strong></div>
                                <hr/>
                                <div className="messages">
                                {this.state.messages.map(message => {
                                        return (
                                            
                                            <div className="message"> <div><img src={this.props.logo} className={this.props.logoStyle}/></div> <strong>{message.author}</strong>: {message.message}</div>
                                     
                                        )
                                    })}
                                    
                                </div>

                            </div>
                            <div className="card-footer">
                                <input type="text" placeholder="Message" className="form-control" value={this.state.message} onChange={ev => this.setState({message: ev.target.value})}/>
                                <br/>
                                <button onClick={this.sendMessage} className="btn btn-primary form-control">Send</button>
                            </div>
                        </div>
                    </div>
                </div>
            </div>
            </form>
            <div className="compNavigation">
            <Navigation />
            </div>
            </div>
        );
    }
}

const mapStateToProps = (state) => { 
    return { 
      user: state.auth.user ,
      isAuthenticated: state.auth.isAuthenticated
      };
  };
  
  export default connect(mapStateToProps,null)(Chat);