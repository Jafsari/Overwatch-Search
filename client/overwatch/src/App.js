import React, { Component } from 'react';
import Landing from './Containers/Landing.jsx'
import { Route, BrowserRouter, Switch} from 'react-router-dom'
import Dashboard from './Containers/Dashboard.jsx'
import requireAuth from './Containers/Auth.jsx'
import Signup from './Containers/Signup.jsx'
import Chat from './Containers/chatroom.jsx'
import Login from './Containers/Login.jsx'
import Diamond from './Containers/Diamond.jsx'
import Master from './Containers/Master.jsx'
import Grandmaster from './Containers/Grandmaster.jsx'
import General from './Containers/General.jsx'
import Search from './Containers/Search.jsx'
import Playerinfo from'./Containers/PlayerInfo.jsx'
import logo from './logo.svg';
import './App.css';

class App extends Component {
  render() {
    return (
      <BrowserRouter>
      <div className="Dash">
        <Switch>
        <Route exact path='/' component={Landing} />
        <Route exact path="/dashboard" component={Dashboard} />
        <Route exact path="/signup" component = {Signup} />
        <Route exact path="/login" component ={Login} />
        <Route exact path="/competitive" component={Chat} />
        <Route exact path="/general" component={General} />
        <Route exact path="/grandmaster" component={Grandmaster} />
        <Route exact path="/master" component={Master} />
        <Route exact path="/diamond" component={Diamond} />
        <Route exact path="/search" component={Search} />
        <Route exact path="/player" component={Playerinfo} />
        </Switch>
     
      </div>
      </BrowserRouter>
    );
  }
}

export default App;
