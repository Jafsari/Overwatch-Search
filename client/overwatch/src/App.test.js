import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { shallow, mount, render } from 'enzyme';
import { createStore,applyMiddleware, compose } from 'redux';
import rootReducer from './reducers';
import App from './App';
import Login from './containers/Login.jsx';
import Modal from './components/Modal.jsx';
import { loadState, saveState } from './localStorage.js';
import reduxThunk from 'redux-thunk';
import Chat from './containers/Chat.jsx';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});


describe('Login Component renders correctly', () => {  
  const persistedState = loadState();
  const store = createStore(
  rootReducer,
  persistedState,
  compose(
    applyMiddleware(reduxThunk),
    window.devToolsExtension ? window.devToolsExtension() : f => f
  )
);
const tree = renderer.create(<Login store={store} />).toJSON();
expect(tree).toMatchSnapshot();
});

describe('Modal Component renders Correctly', () => {
  const tree = renderer.create(<Modal />).toJSON();
  expect(tree).toMatchSnapshot();
});

describe('Chat renders correctly', () => {
  const persistedState = loadState();
  const store = createStore(
  rootReducer,
  persistedState,
  compose(
    applyMiddleware(reduxThunk),
    window.devToolsExtension ? window.devToolsExtension() : f => f
  )
);
  const tree = renderer.create(<Chat store={store} />).toJSON();
  expect(tree).toMatchSnapshot();
})







