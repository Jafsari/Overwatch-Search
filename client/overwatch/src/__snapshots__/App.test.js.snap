// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<div
  className="login"
>
  <form
    className="login2"
  >
    <div>
      <div
        className="form-group"
        onSubmit={[Function]}
      >
        <label
          className=""
          htmlFor="exampleEmail"
        >
          Username
        </label>
        <input
          className="form-control"
          id="username"
          name="username"
          onChange={[Function]}
          placeholder="Enter username"
          type="text"
          value=""
        />
      </div>
      <div
        className="form-group"
      >
        <label
          className=""
          htmlFor="examplePassword"
        >
          Password
        </label>
        <input
          className="form-control"
          id="password"
          name="password"
          onChange={[Function]}
          placeholder="Enter password"
          type="password"
          value=""
        />
      </div>
    </div>
    <div>
      <button
        className="btn btn-danger"
        onClick={[Function]}
        type="button"
      >
        Login
      </button>
    </div>
  </form>
</div>
`;

exports[` 2`] = `
<div>
  <button
    className="btn btn-info"
    onClick={[Function]}
    type="button"
  >
     More Info
  </button>
</div>
`;

exports[` 3`] = `
<div
  className="chat"
>
  <form
    onSubmit={[Function]}
  >
    <img
      alt="logo"
      className={undefined}
      src={undefined}
    />
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="col-8"
        >
          <div
            className="card"
          >
            <div
              className="card-body"
            >
              <div
                className="card-title"
              >
                <strong />
              </div>
              <hr />
              <div
                className="messages"
              />
            </div>
            <div
              className="card-footer"
            >
              <input
                className="form-control"
                onChange={[Function]}
                placeholder="Message"
                type="text"
                value=""
              />
              <br />
              <button
                className="btn btn-primary form-control"
                onClick={[Function]}
              >
                Send
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </form>
  <div
    className="compNavigation"
  >
    <div>
      <nav
        className="navbar navbar-expand-md navbar-light bg-light"
      >
        <a
          className="navbar-brand"
          href="/search"
          onClick={[Function]}
        >
          Search
        </a>
        <button
          className="navbar-toggler"
          onClick={[Function]}
          type="button"
        >
          <span
            className="navbar-toggler-icon"
          />
        </button>
        <div
          className="collapse navbar-collapse"
          style={Object {}}
        >
          <ul
            className="ml-auto navbar-nav"
          >
            <li
              className="nav-item"
            >
              <a
                className="nav-link"
                href="/general"
                onClick={[Function]}
              >
                General Chat
              </a>
            </li>
            <li
              className="nav-item"
            >
              <a
                className="nav-link"
                href="/competitive"
                onClick={[Function]}
              >
                Competitive
              </a>
            </li>
            <li
              className="dropdown nav-item"
              onKeyDown={[Function]}
            >
              <a
                aria-expanded={false}
                aria-haspopup={true}
                className="dropdown-toggle nav-link"
                href="#"
                onClick={[Function]}
              >
                Options
              </a>
              <div
                aria-hidden={true}
                className="dropdown-menu"
                role="menu"
                tabIndex="-1"
                x-placement={undefined}
              >
                <a
                  className="dropdown-item"
                  href="/grandmaster"
                  onClick={[Function]}
                  tabIndex="0"
                  type={undefined}
                >
                  GrandMasters
                </a>
                <a
                  className="dropdown-item"
                  href="/master"
                  onClick={[Function]}
                  tabIndex="0"
                  type={undefined}
                >
                  Masters
                </a>
                <button
                  className="dropdown-item"
                  onClick={[Function]}
                  tabIndex="0"
                  type="button"
                />
                <a
                  className="dropdown-item"
                  href="/diamond"
                  onClick={[Function]}
                  tabIndex="0"
                  type={undefined}
                >
                  Diamond
                </a>
                <div
                  className="dropdown-divider"
                  onClick={[Function]}
                  tabIndex="-1"
                  type={undefined}
                />
                <a
                  className="dropdown-item"
                  href="/dashboard"
                  onClick={[Function]}
                  tabIndex="0"
                  type={undefined}
                >
                  Home
                </a>
              </div>
            </li>
          </ul>
        </div>
      </nav>
    </div>
  </div>
</div>
`;
